.pre-commit-config.yaml

```diff
@@ -28,11 +28,9 @@ repos:
         # Keep exclude in sync with mypy own excludes
         exclude: ^tests/test_data/
         additional_dependencies:
-          - click==8.0.1
-          - pep517==0.10.0
-          - toml==0.10.2
-          - pip==20.3.4
-          - build==0.9.0
+          - click==8.1.6
+          - pip==23.2
+          - build==0.10.0
```

Comment:
We haven't got these requirements pinned in `pyproject.toml` - Should we do this, or instead remove the boundary here?